stdlib/public/Concurrency/TaskPrivate.h

```diff
@@ -238,12 +238,18 @@ class alignas(sizeof(void*) * 2) ActiveTaskStatus {
   bool isStoredPriorityEscalated() const {
     return Flags & IsEscalated;
   }
+
+  ActiveTaskStatus withPriority(JobPriority priority) const {
+    return ActiveTaskStatus(Record,
+                            (Flags & ~PriorityMask) | uintptr_t(priority));
+  }
```

Comment:
Since this is dangerous to use on a running task, it'd be good to rename this to be less tempting (`withPriority_suspended`?  We've used the `_precondition` convention elsewhere, and it stands out nicely) and give it a comment explaining that.

I think it would be good to (1) add a comment saying that this should only be used in circumstances where we know the task isn't running and (2) rename it something less prone to someone reaching for the simpler thing (maybe `withPriority_suspended`?).

Reply:
I renamed it `withNewPriority` and added some comments. I felt a bit uneasy with the word `suspended` so tried to keep it more neutral.